lib/sycamore/sycamore/query/execution/aggregation.py

```diff
@@ -0,0 +1,90 @@
+from collections.abc import Callable
+from typing import Any, Optional
+
+from ray.data import Dataset
+from ray.data.aggregate import AggregateFn
+from ray.data.grouped_data import GroupedData
+
+from sycamore.data import Document
+from sycamore.docset import DocSet
+from sycamore.plan_nodes import Scan
+
+
+# Todo support multiple fields
+def make_aggregation_map_fn(field: str, default_val: Any):
+    def ray_callable(input_dict: dict[str, Any]) -> dict[str, Any]:
+        doc = Document.from_row(input_dict)
+        if field not in doc.properties:
+            agg_val = default_val
+        else:
+            agg_val = doc.properties[field]
+
+        new_doc = doc.to_row()
+        new_doc[field] = agg_val
+
+        return new_doc
+
+    return ray_callable
+
+
+def accumulate_row(doc, doc_bytes, fn):
+    # print(f"In accumulate_row: {doc} {doc_bytes}")
+    return fn(Document.deserialize(doc), Document.deserialize(doc_bytes["doc"])).serialize()
+
+
+def merge(x, y, fn):
+    # print(f"In merge: {Document.deserialize(x)} {Document.deserialize(y)}")
+    return fn(Document.deserialize(x), Document.deserialize(y)).serialize()
+
+
+# Most general version. Keeps everything as Documents. Lots of unnecessary serialization/deserializaiton.
+class DocSetAggregate(AggregateFn):
```

Comment:
Hey, I recognize this code :). I think this should eventually live in the core library, but I gather that's coming in a separate PR.

Reply:
Yep.